README.md

```diff
@@ -18,27 +18,32 @@ Based on [redux-saga-rxjs](https://github.com/salsita/redux-saga-rxjs) by Salsit
 npm install store-saga --save
 ```
 
-Write a saga:
+Write a saga (you _don't_ have to import and specify all those types, but they help to spot errors earlier):
+
 ```ts
+import {Observable} from 'rxjs/Observable';
+import {Action} from '@ngrx/store';
 import {createSaga} from 'store-saga';
 
-export const increment = createSaga(function(){
-  return iteration$ => iteration$
-    .filter(iter => iter.action.type === 'DECREMENT')
-    .map(() => {
-      return { type: 'INCREMENT'}
-    });
+export const incrementProvider = createSaga(function() {
+  return function incrementSaga(iteration$): Observable<Action> {
+    return iteration$
+      .filter(iter => iter.action.type === 'DECREMENT')
+      .map(() => {
```

Comment:
Hello there. In the team we tried working on our code with Visual Studio Code: things look better there, no spurious red squiggles. 

`map` seems fine without the type specifier:

![map-no-type](https://cloud.githubusercontent.com/assets/3185573/14911837/511d0684-0df8-11e6-82e7-16b7776ade2b.png)

and adding the type specifier seems to add nothing:

![map-with-type](https://cloud.githubusercontent.com/assets/3185573/14911858/81c6536c-0df8-11e6-9795-70ad72a5ff0b.png)

I think it makes sense, when I look at `map` type definition:

```js
declare module '../../Observable' {
    interface Observable<T> {
        map: MapSignature<T>;
    }
}
// ... in imported file:
export interface MapSignature<T> {
    <R>(project: (value: T, index: number) => R, thisArg?: any): Observable<R>;
}
```

as it is the Observable that "receives" (so to say) a type. Once that is set, the other "loose end" (`<R>`) gets fixed by the type returned by projection function.

What do you think about it? If you think it's needed anyway, no probs: we can add that so to move on with the PR.